node.go

```diff
@@ -514,6 +520,20 @@ func (n *Node) switchToCandidate() {
 	n.switchState(CANDIDATE)
 }
 
+// Execute `f` in a separate go routine, but ensures that functions
+// are executed in order.
+func (n *Node) serializeGoRoutine(nextCh *(chan bool), f func()) {
```

Comment:
I hear you, just concerned that we have N Go routines that are lock stepped instead of one that is selecting from multiple channels. But your solution may be better.